daemon/api_systems.go

```diff
@@ -111,9 +107,46 @@ func postSystemsAction(c *Command, r *http.Request, user *auth.UserState) Respon
 	if decoder.More() {
 		return BadRequest("extra content found in request body")
 	}
-	if req.Action != "do" {
+	switch req.Action {
+	case "do":
+		return postSystemActionDo(c, systemLabel, &req)
+	case "reboot":
+		return postSystemActionReboot(c, systemLabel, &req)
+	default:
 		return BadRequest("unsupported action %q", req.Action)
 	}
+}
+
+// XXX: should deviceManager return more sensible errors here?
```

Comment:
a followup maybe?

Reply:
Yeah, I thin this is big enough already.